src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -2492,6 +2492,104 @@ void QgsFieldDomainItemGuiProvider::populateContextMenu( QgsDataItem *item, QMen
       }
     }
   }
+
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION( 3, 12, 0 )
```

Comment:
Can this if branch be merged into the above branch, so that we avoid the duplicated code for providerKey/connectionUri retrieval? The GDAL version check should not be required as we should be reading the capabilities from the connection.